tools/prepare_lava_job.py

```diff
@@ -0,0 +1,169 @@
+will be used with our RevPi devices in order to flash and perform tests
+with LAVA automatically, thus saving time finding errors in the images.
+"""
+
+import argparse
+import os
+import pkg_resources
+import subprocess
+import sys
+import yaml
+
+NEW_JOB_NAME = "job_flash_test.yaml"
+LAVACLI_JOB_SUBMIT = "lavacli jobs submit "
+
+parser = argparse.ArgumentParser(
+    description=__doc__,
+    epilog="Default behaviour: ...",
+)
+
+parser.add_argument(
+    "-d",
+    "--device-type",
+    dest="device_type",
+    nargs="?",
+    help="name of device type to submit job. Default: RevPi_Core",
+)
+
+parser.add_argument(
+    "-u",
+    "--url",
+    dest="url",
+    help="URL to the image",
+)
+
+parser.add_argument(
+    "--flash",
+    dest="flash",
+    nargs="?",
+    required=True,
+    help="file used to flash image",
+)
+
+parser.add_argument(
+    "-s",
+    "--testsuite",
+    dest="testsuite",
+    required=True,
+    help="job testsuite to be used after flashing",
+)
+
+parser.add_argument(
+    "-l",
+    "--lava",
+    dest="lava",
+    type=str,
+    nargs='?',
+    const="True",
+    help="submit generated job with lavacli",
+)
+
+def run_lavacli() -> bool:
+    try:
+        pkg_resources.get_distribution("lavacli")
+    except pkg_resources.DistributionNotFound:
+        print("lavacli is not installed...")
+        return False
+    
+    lavacli_cmd = LAVACLI_JOB_SUBMIT + NEW_JOB_NAME
+    process = subprocess.Popen(lavacli_cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    stdout = stdout.decode('utf-8')
+    stderr = stderr.decode('utf-8')
+    
+    print(f"{lavacli_cmd}")
+    print(stdout)
+
+    if stderr:
+        print("lavacli Error:")
+        print(stderr)
+        return False
+
+    return True
+
+#TODO what about another URLs???
+def get_imagename_from_url(url):
+    return os.path.basename(url)
+
+def add_namespace_in_testsuite(content):
+    actions = content.get("actions", [])
+
+    for action in actions:
+        if "test" in action or "boot" in action or "deploy" in action:
+            action_name = next(iter(action))
+            action[action_name]["namespace"] = "testsuite"
+
+    return content
+
+def create_job(device_type, image_url, file_flash, file_testsuite) -> bool:
+    print("Generate Job... ", end="")
+
+    # Load the contents of file_flash
+    try:
+        with open(file_flash, 'r') as file:
+            file_flash_contents = yaml.safe_load(file)
+    except OSError:
+        print(f"\nCould not read file to flash device: {file_flash}")
+        return False
+    print(file_testsuite)
+    # Load the contents of file_testsuite
+    try:
+        with open(file_testsuite, 'r') as file:
+            file_testsuite_contents = yaml.safe_load(file)
+    except OSError:
+        print(f"\nCould not read file with testsuite: {file_testsuite}")
+        return False
+    print("blablabla... ")
```

Comment:
Not sure if you mean for this to still be here :p

Reply:
Thanks...